### ISSUE TITLE
"preferred way" example in "working with conditions" produces typescript error

### ISSUE THREAD

**Issue by tsnieman** [raycharius/slack-block-builder#61]
**Describe the Bug**
The ["preferred way" documentation in "working with conditions"](https://www.blockbuilder.dev/#/working-with-conditions?id=preferred-way) does not seem to be a valid example — in practice, the example produces a TypeScript error

Note:
> ⚠️ **I'm not entirely sure if this is a bug or a documentation issue**

**To Reproduce**
Using v2.0.1, try this:

```ts
import { Blocks, Modal } from 'slack-block-builder';

const someData = undefined;
const example = Modal({ title: 'Using If Statements' })
    .blocks(
        Blocks.Section({ text: 'Let\'s take a look at conditionals.' }),
        Blocks.Section({ text: 'This is a much better way to do it!' }))
    .blocks(someData && [
        Blocks.Section({ text: 'This is added if condition is true.' }),
        Blocks.Section({ text: 'And it\'s super easy and readable!' }),
    ])
```

<img width="862" alt="Screen Shot 2021-07-06 at 2 15 52 PM" src="https://user-images.githubusercontent.com/595711/124661426-b59d1a80-de64-11eb-8833-f6cfe3730ef6.png">

Given that I don't know what `someData` is, I'm assuming it would be either `undefined` or a truthy value, and given that, below the example in the docs, it is noted:
> For conditions that provide a `true` or `false` instead of `undefined`, using a ternary operator is a great way to go.

I'm lead to believe that `undefined` should be a valid value for `someData`, and that `.blocks(undefined)` would also be valid.

Here's the (also not working) ternary version of the example:
```ts
const someData = undefined;
const example = Modal({ title: 'Using If Statements' })
    .blocks(
        Blocks.Section({ text: 'Let\'s take a look at conditionals.' }),
        Blocks.Section({ text: 'This is a much better way to do it!' }))
    .blocks(someData ? [
        Blocks.Section({ text: 'This is added if condition is true.' }),
        Blocks.Section({ text: 'And it\'s super easy and readable!' }),
    ]: undefined)
```

<img width="855" alt="Screen Shot 2021-07-06 at 2 15 22 PM" src="https://user-images.githubusercontent.com/595711/124661340-9d2d0000-de64-11eb-8167-596a895b34cc.png">

Though if I return an empty array in the ternary (instead of `undefined`), that is an acceptable value.

**Expected Behavior**

Given that I'm unsure if this is a documentation issue or actual bug:

- the documentation should provide a clearer/valid example **and/or** the typedefs should be updated to reflect valid values

**Additional Context**
n/a

**Comment by raycharius** [comment]
Thanks, @tsnieman! This is working for me. Could you share your tsconfig and TS version? If this isn't working, it's definitely a bug.

**Comment by tsnieman** [comment]
Following up on my previous hunch about the TypeScript version ⬆️ I upgraded to the latest TS as a test, but the original example still results in the same error.

**Comment by raycharius** [comment]
I've just released this in version 2.1.0 – let me know if you run into any other issues!

**Comment by tsnieman** [comment]
`undefined` works now - thanks @raycharius :+1:
